### DISCUSSION TITLE
Subscriptions aren't getting acknowledged in Kubernetes Production environment

### DISCUSSION THREAD

**Discussion by Danieth** [anycable/anycable#157] [Q&A] [answered]
Hey folks,

I've exhausted my resources here trying to figure out why I can't actually acquire a subscription to a channel, exclusively, in my production environment.

I'm deploying on DigitalOcean, with a very simple nginx configuration - which, has, had some issues websockets - however, I don't believe the connection itself is being severed in any way, primarily, because the client is receiving ping messages back.

It appears to me that, for some reason, RPC commands are just not hitting the rpc server in the Kubernetes environment vs run locally.

## Javascript

I am using the AnyCable JS client (previously, the Rails ActionCable client -- same result)

### Javascript Code
<img width="624" alt="CleanShot 2022-07-26 at 01 15 20@2x" src="https://user-images.githubusercontent.com/4342908/180936466-050e4e52-03ba-4529-aa77-2c2f8cb302ba.png">

### Production (Failing)
<img width="901" alt="CleanShot 2022-07-26 at 00 41 05@2x" src="https://user-images.githubusercontent.com/4342908/180931803-3a7a388b-4c51-4b1c-88f3-79f7fc1c7b91.png">

### Local Development (Succeeding)
<img width="922" alt="CleanShot 2022-07-26 at 00 51 18@2x" src="https://user-images.githubusercontent.com/4342908/180933054-08e16fae-a852-4e36-9292-699262ddd82d.png">

## Network

In the network tab, it's clear, I'm receiving ping's from the `anycable-go` server!

### Production (Failing)
<img width="730" alt="CleanShot 2022-07-26 at 00 47 57@2x" src="https://user-images.githubusercontent.com/4342908/180932601-9f4ffeb2-de93-423e-9a8a-6d3c0f7f0ca6.png">

### Local Development (Succeeding)
<img width="694" alt="CleanShot 2022-07-26 at 00 48 22@2x" src="https://user-images.githubusercontent.com/4342908/180932667-d7c181fd-e4d2-46c3-92a3-41aca8ec4f65.png">

## Rails Code

There's nothing special happening on the backend. As far as I can tell (based on print out statements) the channel code is never actually being called in `Production`

### Rails Channel Code
<img width="484" alt="CleanShot 2022-07-26 at 01 25 02@2x" src="https://user-images.githubusercontent.com/4342908/180938018-0e95492c-74c0-4f9f-a5ed-6f50e61b9d08.png">

### Rails Cable Code
<img width="464" alt="CleanShot 2022-07-26 at 01 25 44@2x" src="https://user-images.githubusercontent.com/4342908/180938106-06cef5ab-e6d9-4516-abc1-d9cf6484d3a9.png">

## Sever Logs

These are the Rails server logs, just for a single attempt to connect to the Server.

### Production, Kubernetes instances (Failing)

Instances are roughly a 1:1 with the suggested helm file document: https://docs.anycable.io/deployment/kubernetes

`stern anycable -t --since 10m`
![180933476-3fc3ca09-f7fe-413e-a7fd-aa287dc1b853](https://user-images.githubusercontent.com/4342908/180934870-2f1a4cc2-5036-492e-a0b6-d5c4616901a3.png)

### Local Development (Succeeding)

Procfile:

```
anycable-rpc: bundle exec anycable --log-grpc --debug
anycable-go:  anycable-go --port 8080 --log_level=debug
```
<img width="1680" alt="CleanShot 2022-07-26 at 00 56 47@2x" src="https://user-images.githubusercontent.com/4342908/180935033-e214d42e-b11b-4e99-b85e-454a731443bf.png">

## Redis

I confirmed with (connecting to dev and prod redis instances):

```
$redis.subscribe("__anycable__") do |on|
  on.subscribe do |channel, subscriptions|
    puts "Subscribed to ##{channel} (#{subscriptions} subscriptions)"
  end
  on.message do |channel, message|
    puts "##{channel}: #{message}"
    redis.unsubscribe if message == "exit"
  end
  on.unsubscribe do |channel, subscriptions|
    puts "Unsubscribed from ##{channel} (#{subscriptions} subscriptions)"
  end
end
```

that my messages are actually passing through the `__anycable__` key for both development and production.

**Comment by Danieth** [comment, marked as answer]
Unfortunately, this was simply a manifestation of CloudFlares "websocket" option not being turned on

<img width="1027" alt="CleanShot 2022-07-28 at 02 07 52@2x" src="https://user-images.githubusercontent.com/4342908/181443014-1ccb958e-c571-43b1-93b2-d1e383a6ece4.png">

I simply, did not, realize that this setting had been disabled! Thanks!

**Reply by Envek** [reply]
Thanks for sharing the solution and for detailed description of the problem!
